lib/messages.coffee

```diff
@@ -0,0 +1,81 @@
+Helpers = require('./helpers')
+{CompositeDisposable, Emitter} = require('atom')
+
+###
+  Note: We are reclassifying the messages on on Pane Item Change,
+  even though we are relinting on that same event, 'cause linters
+  could take time and we have to refresh the views immediately
+###
+
+class MessageRegistry
+  constructor: (@linter)->
+    @count = File: 0, Project: 0
+    @messages = new Map() # Messages = Map<Linter, Array<messages>>
+    @emitter = new Emitter
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.workspace.onDidChangeActivePaneItem =>
+      @messages.forEach (messages) => @classifyMessages(messages)
+      @emitter.emit 'did-classify'
+
+  set: (linter, messages) ->
+    Helpers.validateMessages(messages)
+    @classifyMessages(messages)
```

Comment:
Well, 'did-classify' is also called on line 18 where 'did-update' is not called.